src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1327,61 +1327,99 @@ public override BoundNode VisitConditionalOperator(BoundConditionalOperator node
                 Unsplit();
-                var alternativeType = _result.Type;
                 IntersectWith(ref this.State, ref consequenceState);
-                if ((object)consequenceType == null || (object)alternativeType == null || !consequenceType.Equals(alternativeType, TypeCompareKind.ConsiderEverything))
+                resultIsNullable = (getIsNullable(consequence, consequenceResult) | getIsNullable(alternative, alternativeResult));
+            }
+
+            TypeSymbolWithAnnotations resultType;
+            if (node.HasErrors)
+            {
+                resultType = null;
+            }
+            else
+            {
+                HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+                resultType = BestTypeInferrer.InferBestTypeForConditionalOperator(
+                    createPlaceholderIfNecessary(consequence, consequenceResult),
+                    createPlaceholderIfNecessary(alternative, alternativeResult),
+                    _conversions,
+                    out _,
+                    ref useSiteDiagnostics);
+                if (resultType is null)
                 {
-                    var consequenceIsNullable = (object)consequenceType == null ? true : consequenceType.IsNullable;
-                    var alternativeIsNullable = (object)alternativeType == null ? true : alternativeType.IsNullable;
-                    _result = TypeSymbolWithAnnotations.Create(node.Type, isNullableIfReferenceType: consequenceIsNullable | alternativeIsNullable);
+                    ReportStaticNullCheckingDiagnostics(
+                        ErrorCode.WRN_NoBestNullabilityConditionalExpression,
+                        node.Syntax,
+                        GetTypeAsDiagnosticArgument(consequenceResult.Type?.TypeSymbol),
+                        GetTypeAsDiagnosticArgument(alternativeResult.Type?.TypeSymbol));
                 }
-                else
+            }
+            resultType = TypeSymbolWithAnnotations.Create(resultType?.TypeSymbol ?? node.Type, resultIsNullable);
+
+            _result = resultType;
+            return null;
+
+            bool? getIsNullable(BoundExpression expr, Result result)
+            {
+                var type = result.Type;
+                if ((object)type != null)
                 {
-                    // PROTOTYPE(NullableReferenceTypes): Use BestTypeInferrer.InferBestTypeForConditionalOperator
-                    // to ensure nested nullability is considered.
-                    _result = GetMoreNullableType(consequenceType, alternativeType);
+                    return type.IsNullable;
                 }
-
-                // it may not be a boolean state at this point (5.3.3.28)
-                // PROTOTYPE(NullableReferenceTypes): Report conversion warnings.
+                if (expr.IsLiteralNullOrDefault())
+                {
+                    return true;
```

Comment:
>return true; [](start = 20, length = 12)

I do not know if that can have any negative effect, but default of a value type is not nullable.

Reply:
This value is used for `isNullableIfReferenceType` parameter value. I'll rename the local function and `resultIsNullable` local in a subsequent change.